src/flb_config.c

```diff
@@ -309,6 +309,11 @@ struct flb_config *flb_config_init()
 
     /* Environment */
     config->env = flb_env_create();
+    if (!config->env) {
```

Comment:
Would you mind changing this comparison to `if (config->env == NULL)` if you happen to make the change requested in the other file?

Reply:
yep, will fix once we find a solution to the null-dereference in https://github.com/fluent/fluent-bit/pull/6904#discussion_r1120320160